source/_components/cover.mysensors.markdown

```diff
@@ -0,0 +1,152 @@
+---
+layout: page
+title: "MySensors Cover"
+description: "Instructions how to integrate MySensors covers into Home Assistant."
+date: 2016-09-25 11:30 +0100
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: mysensors.png
+ha_category: Cover
+ha_release: 0.29
+---
+
+Integrates MySensors covers into Home Assistant. See the [main component] for configuration instructions.
+
+The following actuator types are supported:
+
+##### MySensors version 1.4
+S_TYPE      | V_TYPE
+------------|-------------
+S_COVER     | V_UP, V_DOWN, V_STOP, [V_DIMMER or V_LIGHT]
+
+##### MySensors version 1.5 and higher
+S_TYPE      | V_TYPE
+------------|-------------
+S_COVER     | V_UP, V_DOWN, V_STOP, [V_PERCENTAGE or V_STATUS]
+
+All V_TYPES above are required. Use V_PERCENTAGE (or V_DIMMER) if you know the exact position of the cover in percent, use V_STATUS (or V_LIGHT) if you don't.
+
+For more information, visit the [serial api] of MySensors.
+
+### {% linkable_title Example sketch %}
+
+```cpp
+/*
+ * Documentation: http://www.mysensors.org
+ * Support Forum: http://forum.mysensors.org
+ */
+
+// Enable debug prints to serial monitor
+#define MY_DEBUG
+#define MY_RADIO_NRF24
+
+#include <MySensors.h>
+#define SN "Cover"
+#define SV "1.1"
+
+// Actuators for moving the cover up and down respectively.
+#define COVER_UP_ACTUATOR_PIN 2
+#define COVER_DOWN_ACTUATOR_PIN 3
+// Sensors for finding out when the cover has reached its up/down position.
+// These could be simple buttons or linear hall sensors.
+#define COVER_UP_SENSOR_PIN 4
+#define COVER_DOWN_SENSOR_PIN 5
+
+#define CHILD_ID 0
+
+// Internal representation of the cover state.
+enum State {
+  IDLE,
+  UP, // Window covering. Up.
+  DOWN, // Window covering. Down.
+};
+
+static int state = IDLE;
+static int status = 0; // 0=cover is down, 1=cover is up
+MyMessage upMessage(CHILD_ID, V_UP);
+MyMessage downMessage(CHILD_ID, V_DOWN);
+MyMessage stopMessage(CHILD_ID, V_STOP);
+MyMessage statusMessage(CHILD_ID, V_STATUS);
+
+void sendState() {
+  // Send current state and status to gateway.
+  send(upMessage.set(state == UP));
+  send(downMessage.set(state == DOWN));
+  send(stopMessage.set(state == IDLE));
+  send(statusMessage.set(status));
+}
+
+void setup() {
+  pinMode(COVER_UP_SENSOR_PIN, INPUT);
+  pinMode(COVER_DOWN_SENSOR_PIN, INPUT);
+
+  // Send initial values.
+  sendState();
```

Comment:
In mysensors 2.0.0 this has to be done in the loop, due to registration on gateway required before device can send values. This has changed in dev branch of mysensors, but that's not released yet.